heartbeat/podman-etcd

```diff
@@ -1078,11 +1078,25 @@ compare_revision()
 	revision=$(attribute_node_revision get)
 	peer_revision=$(attribute_node_revision_peer)
 
-	if [ "$revision" = "" ] || [ "$revision" = "null" ] || [ "$peer_revision" = "" ] || [ "$peer_revision" = "null" ]; then
+	if [ "$revision" = "" ] || [ "$revision" = "null" ] && [ "$peer_revision" = "" ] || [ "$peer_revision" = "null" ]; then
 		ocf_log err "could not compare revisions: $NODENAME local revision: $revision, peer revision: $peer_revision"
 		return "$OCF_ERR_GENERIC"
 	fi
 
+	# When only the local revision exists, it can start normally. This ensures that it can start during node replacement events.
+	if [ "$peer_revision" = "" ] || [ "$peer_revision" = "null" ]; then
+		ocf_log info "only local revision exists: $NODENAME local revision: $revision, peer revision: $peer_revision"
+		echo "only_local"
+		return "$OCF_SUCCESS"
+	fi
+
+	# When only the peer revision exists, the local node must join as a learner. This ensures that it can start during node replacement events.
```

Comment:
This is less risky, as the local revision comes directly from disk, so this might occur only if the node has no etcd data at all.